Apromore-Custom-Plugins/REST-Endpoint/src/main/java/org/apromore/rest/manager/ArtifactResource.java

```diff
@@ -318,10 +322,23 @@ private String findAttribute(final String name) {
         return values == null ? null : values.stream().findFirst().orElse(null);
     }
 
+    /**
+     * @param name  an HTTP header name
+     * @return all values of the <var>name</var>d header occurring within {@link #httpHeaders}, honoring both
+     *     duplicate headers and headers with comma-delimited lists of values
+     */
     private List<String> findAttributes(final String name) {
         List<String> requestHeaders = httpHeaders.getRequestHeader(name);
 
-        return requestHeaders == null ? Collections.emptyList() : requestHeaders;
+        if (requestHeaders == null) {
```

Comment:
Seems to be redundant check

Reply:
The getRequestHeader method returns null rather than an empty list, so the check is necessary to guarantee the method always returns a List.